e2e/cypress/tests/integration/settings/display/message_display_mode_colorize_username_spec.js

```diff
@@ -0,0 +1,140 @@
+        cy.findByRole('checkbox', {
+            name: 'Colorize usernames: Use colors to distinguish users in compact mode',
+        }).should('not.exist');
+
+        // # Save and close the modal
+        cy.uiSave();
+        cy.uiClose();
+    });
+
+    it('MM-T4984_2 Message Display: colorize usernames option should exist in Compact mode and function as expected', () => {
+        // # Select 'Compact' option
+        cy.findByRole('heading', {name: 'Message Display'}).click();
+        cy.findByRole('radio', {name: 'Compact: Fit as many messages on the screen as we can.'}).click();
+
+        // * Verify Colorize usernames option exists and checked by default
+        cy.findByRole('checkbox', {name: 'Colorize usernames: Use colors to distinguish users in compact mode'}).should('exist');
+        cy.findByRole('checkbox', {name: 'Colorize usernames: Use colors to distinguish users in compact mode'}).should('be.checked');
+
+        // # Save and close the modal
+        cy.uiSave();
+        cy.uiClose();
+
+        // # Save the color of the buttons
+        cy.findByText(firstUser.username).then((elements) => {
+            colors[firstUser.username] = elements[0].attributes.style.value;
+        });
+        cy.findByText(otherUser.username).then((elements) => {
+            colors[otherUser.username] = elements[0].attributes.style.value;
+        }).then(() => {
+            // * Verify that colors are different
+            expect(colors[firstUser.username]).to.not.equal(colors[otherUser.username]);
+        });
+
+        cy.reload();
+
+        // * Verify that after reload colors are the same
+        cy.findByText(firstUser.username).then((elements) => {
+            cy.wrap(elements[0]).should('have.attr', 'style', colors[firstUser.username]);
+        });
+        cy.findByText(otherUser.username).then((elements) => {
+            cy.wrap(elements[0]).should('have.attr', 'style', colors[otherUser.username]);
+        });
+    });
+
+    it('MM-T4984_3 Message Display: disabling colorize should revert to colors to normal color', () => {
+        // # Select 'Compact' option
+        cy.findByRole('heading', {name: 'Message Display'}).click();
+        cy.findByRole('radio', {name: 'Compact: Fit as many messages on the screen as we can.'}).click();
+
+        // # Verify Colorize usernames option exists and make it unchecked
+        cy.findByRole('checkbox', {name: 'Colorize usernames: Use colors to distinguish users in compact mode'}).should('exist');
+        cy.findByRole('checkbox', {name: 'Colorize usernames: Use colors to distinguish users in compact mode'}).uncheck().should('not.be.checked');
+
+        // # Save and close the modal
+        cy.uiSave();
+        cy.uiClose();
+
+        // * Verify that colors are reverted to normal
+        cy.findByText(firstUser.username).then((elements) => {
+            cy.wrap(elements[0]).should('have.css', 'color', 'rgb(63, 67, 80)');
```

Comment:
I didn't like to write color hard coded, i.e., `rgb(63, 67, 80)`. I searched in the codebase, but I couldn't find any global variables for the color. 

Please let me know if there is a global variable, so I can replace `rgb(63, 67, 80)`

Reply:
> I didn't like to write color as hard coded, i.e., `rgb(63, 67, 80)`. I searched in the codebase, but I couldn't find any global variables for the color.
> 
> Please let me know if there is a global variable, so I can replace `rgb(63, 67, 80)`

you may look into this test
e2e/cypress/tests/integration/settings/display/theme/custom_theme_sidebar_styles_spec.js

Reply:
I had a look but this file is still using it as hard coded. Therefore, I left as it is.

Reply:
An idea to avoid hardcoding the color would be to retrieve it at the beginning of this testcase, before enabling Compact mode. E.g. could be done with something similar to this:
```
        cy.findByText(firstUser.username).then((elements) => {
          cy.window().then((win) => {
            defaultColor = win.getComputedStyle(elements[0]).color;
          });
        });
```

Reply:
Nice workaround solution @mvitale1989. It worked perfectly.